Library/Formula/sdl2_mixer.rb

```diff
@@ -20,17 +20,30 @@ class Sdl2Mixer < Formula
   depends_on "flac" => :optional
   depends_on "fluid-synth" => :optional
   depends_on "smpeg2" => :optional
+  depends_on "libmodplug" => :optional
   depends_on "libmikmod" => :optional
   depends_on "libvorbis" => :optional
 
   def install
+    if build.with?("libmodplug") && !build.head? then
+      # libmodplug support was fixed in changeset 695 (https://hg.libsdl.org/SDL_mixer/rev/6a5e6d8d6a35).
+      odie "sdl2_mixer 2.0.0 doesn't properly support a compilation with libmodplug. Use --HEAD."
+    end
```

Comment:
Probably don't want to add the option until it's used by a stable version to avoid confusion.